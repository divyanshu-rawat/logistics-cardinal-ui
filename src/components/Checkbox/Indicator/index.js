import React from 'react';
import PropTypes from 'prop-types';

import IndicatorStyled from './IndicatorStyled';
import Checkbox from '..';

const Indicator = (props) => (
  <IndicatorStyled color={props.color}>
    <Checkbox {...props} />
  </IndicatorStyled>
);

Indicator.propTypes = {
  label: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  initialValue: PropTypes.bool,
  color: PropTypes.string,
};

export default Indicator;
