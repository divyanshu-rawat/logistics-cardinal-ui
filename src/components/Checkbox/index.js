import React, { Fragment, PureComponent } from 'react';
import PropTypes from 'prop-types';

class Checkbox extends PureComponent {
  static propTypes = {
    label: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    initialValue: PropTypes.bool,
  };

  state = {
    selectedValue: this.props.initialValue || false,
  };

  handleChange = (event) => {
    event.persist();

    this.setState({ selectedValue: !this.state.selectedValue }, () =>
      this.props.onChange(event.target.checked),
    );
  };

  render() {
    const { selectedValue } = this.state;
    const { id, label, name } = this.props;

    return (
      <Fragment>
        <input
          id={id}
          name={name}
          type="checkbox"
          onChange={this.handleChange}
          checked={selectedValue}
        />
        <label htmlFor={id}>{label}</label>
      </Fragment>
    );
  }
}

export default Checkbox;
