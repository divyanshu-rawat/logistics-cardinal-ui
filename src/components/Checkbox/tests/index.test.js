import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Checkbox from '..';

const onChangeFn = jest.fn();

const renderedComponent = (value) =>
  shallow(
    <Checkbox
      label="My Checkbox"
      onChange={onChangeFn}
      initialValue={value}
      id="myId"
      name="myName"
    />,
  );

describe('<Checkbox />', () => {
  it('should render the component', () => {
    const tree = toJson(renderedComponent());
    expect(tree).toMatchSnapshot();
  });

  it('should render the component with `initial value`', () => {
    const tree = toJson(renderedComponent(true));
    expect(tree).toMatchSnapshot();
  });

  it('should call `onChangeFn` when value is changed', () => {
    renderedComponent()
      .find('input')
      .simulate('change', { persist: jest.fn(), target: { checked: true } });

    expect(onChangeFn.mock.calls.length).toBe(1);
  });
});
