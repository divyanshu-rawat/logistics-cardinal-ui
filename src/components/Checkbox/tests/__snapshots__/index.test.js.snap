// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render the component 1`] = `
<React.Fragment>
  <input
    checked={false}
    id="myId"
    name="myName"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor="myId"
  >
    My Checkbox
  </label>
</React.Fragment>
`;

exports[`<Checkbox /> should render the component with \`initial value\` 1`] = `
<React.Fragment>
  <input
    checked={true}
    id="myId"
    name="myName"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor="myId"
  >
    My Checkbox
  </label>
</React.Fragment>
`;
